orderapi/src/main/java/fi/hel/verkkokauppa/order/service/order/OrderService.java

```diff
@@ -131,6 +131,17 @@ public Order createByParams(String namespace, String user) {
         return order;
     }
 
+    public Order createByParams(String namespace, String user, LocalDateTime lastValidPurchaseDateTime) {
+        LocalDateTime createdAt = DateTimeUtil.getFormattedDateTime();
+        String orderId = generateOrderId(namespace, user, createdAt);
+        Long incrementId = this.incrementId.generateOrderIncrementId();
+        Order order = new Order(orderId, namespace, user, createdAt, incrementId, lastValidPurchaseDateTime);
+
```

Comment:
Could you add lastValidPurchaseDateTime with setter, using new constructor makes code slower to add new fields and it takes time to change many constructors.

Code could look like 
`
       ....
        if (StringUtils.isNotEmpty(lastValidPurchaseDateTime)) {
            order.lastValidPurchaseDateTime(lastValidPurchaseDateTime);
        }
        orderRepository.save(order);
        log.debug("created new order, orderId: " + orderId);
        return order;
`

Reply:
Yes, it makes much more sense as you pointed out in the Order model comment.